api/src/paths/critter-data/critters/filter.ts

```diff
@@ -0,0 +1,128 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { PROJECT_PERMISSION, SYSTEM_ROLE } from '../../../constants/roles';
+import { authorizeRequestHandler } from '../../../request-handlers/security/authorization';
+import { CritterbaseService, ICritterbaseUser } from '../../../services/critterbase-service';
+import { getLogger } from '../../../utils/logger';
+
+const defaultLog = getLogger('paths/critter-data/critters/filter');
+export const POST: Operation = [
+  authorizeRequestHandler((req) => {
+    return {
+      or: [
+        {
+          validProjectPermissions: [PROJECT_PERMISSION.COORDINATOR, PROJECT_PERMISSION.COLLABORATOR],
+          projectId: Number(req.params.projectId),
+          discriminator: 'ProjectPermission'
+        },
+        {
+          validSystemRoles: [SYSTEM_ROLE.DATA_ADMINISTRATOR],
+          discriminator: 'SystemRole'
+        }
+      ]
+    };
+  }),
+  filterCritters()
+];
+
+POST.apiDoc = {
+  description: 'Retrieves critters by filtering the global list with the criteria specified in the request body.',
+  tags: ['critterbase'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  requestBody: {
+    description: 'Filtering request object',
+    content: {
+      'application/json': {
+        schema: {
+          title: 'Bulk post request object',
+          type: 'object',
+          properties: {
+            critters: {
+              type: 'array',
+              items: {
+                type: 'string'
+              }
+            },
+            animal_ids: {
+              type: 'array',
+              items: {
+                type: 'string'
```

Comment:
I'm assuming all of these ID arrays are UUIDs? If so, we should include `format: 'uuid'`:

```javascript
animal_ids: {
  type: 'array',
  items: {
    type: 'string',
    format: 'uuid'
  }
},
```

Reply:
No UUIDs. Confusing naming convention but that's just a BCTW moment.